gensrc/script/functions.py

```diff
@@ -839,6 +839,12 @@
     [150322, 'array_sortby', 'ANY_ARRAY',  ['ANY_ARRAY', 'ARRAY_DATE'],      'ArrayFunctions::array_sortby_date'],
     [150323, 'array_sortby', 'ANY_ARRAY',  ['ANY_ARRAY', 'ARRAY_JSON'],      'ArrayFunctions::array_sortby_json'],
 
+    [150330, 'array_generate', 'ARRAY_TINYINT',  ['TINYINT', 'TINYINT', 'TINYINT'],          "ArrayFunctions::array_generate_tinyint"],
+    [150331, 'array_generate', 'ARRAY_SMALLINT',  ['SMALLINT', 'SMALLINT', 'SMALLINT'],      "ArrayFunctions::array_generate_smallint"],
+    [150332, 'array_generate', 'ARRAY_INT',  ['INT', 'INT', 'INT'],                          "ArrayFunctions::array_generate_int"],
+    [150333, 'array_generate', 'ARRAY_BIGINT',  ['BIGINT', 'BIGINT', 'BIGINT'],              "ArrayFunctions::array_generate_bigint"],
```

Comment:
why not only return a bigint?

Reply:
just  for saving unnecessay memory usage

Reply:
I agree with @mofeiatwork , at first, we can just support BIGINT, and user can cast to any other types of Array they want.

For example they can do `cast(array_generate(1,10,1) as array<int>)`

Reply:
@imay @mofeiatwork
frankly speak,   bigint sigature of this function can be extended to other integer types with little endeavor, because expr framework  is encapsulated in a well-form.